dev-support/git-jira-release-audit/README.md

```diff
@@ -0,0 +1,186 @@
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Git / JIRA Release Audit
+
+This is an application for performing an audit between the histories on our git
+branches and the `fixVersion` field set on issues in JIRA. It does this by
+building a Sqlite database from the commits found on each git branch,
+identifying Jira IDs and release tags, and then requesting information about
+those issues from Jira. Once both sources have been collected, queries can be
+performed against the database to look for discrepancies between the sources of
+truth (and, possibly, bugs in this script).
+
+## Setup
+
+The system prerequisites are Python3 with VirtualEnv available and Sqlite. Also,
+you'll need the content of this directory and a local checkout of git repository.
+
+Build a VirtualEnv with the script's dependencies with
+
+```shell script
+$ python3 --version
+Python 3.7.6
+$ python3 -m venv ./venv
+$ ./venv/bin/pip install -r ./requirements.txt
+...
+Successfully installed...
+```
+
+## Usage
+
+The tool provides basic help docs.
+
+```shell script
+$ ./venv/bin/python ./git_jira_release_audit.py --help
+usage: git_jira_release_audit.py [-h] [--db-path DB_PATH]
+                                 [--git-repo-path GIT_REPO_PATH]
+                                 [--remote-name REMOTE_NAME]
+                                 [--development-branch DEVELOPMENT_BRANCH]
+                                 [--development-branch-fix-version DEVELOPMENT_BRANCH_FIX_VERSION]
+                                 [--release-line-regexp RELEASE_LINE_REGEXP]
+                                 [--fallback-actions-path FALLBACK_ACTIONS_PATH]
+                                 [--jira-url JIRA_URL] --branch-1-fix-version
+                                 BRANCH_1_FIX_VERSION --branch-2-fix-version
+                                 BRANCH_2_FIX_VERSION
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --db-path DB_PATH     Path to the database file, or leave unspecified for a
+                        transient db.
+  --git-repo-path GIT_REPO_PATH
+                        Path to the git repo, or leave unspecified to infer
+                        from the current file's path.
+  --remote-name REMOTE_NAME
+                        The name of the git remote to use when identifying
+                        branches.
+  --development-branch DEVELOPMENT_BRANCH
```

Comment:
I understand the words you have written, but I'm not sure what I would actually pass to this option. I see in the example below you didn't provide it, but I'm not sure when I would want/need to provide it.

Reply:
So for HBase and 95% of git-tracked projects, this is `master`. I believe Hadoop calls the `trunk`, as per its Subversion legacy. It's the branch where development takes place; I'm not sure what else to call this.

Reply:
Gotcha. Wasn't sure if there was a case where we'd be providing a value for hbase. I understand the hadoop use-case. Thanks!